### PR TITLE
Dump PHP 7.1

### PR DESCRIPTION
**Pull request by alexmanno**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,17 +15,6 @@ env:
 matrix:
   fast_finish: true
   include:
-    - php: 7.0
-      env:
-        - DEPS=lowest
-    - php: 7.0
-      env:
-        - EXECUTE_CS_CHECK=true
```
Comment (thomasvargiu) [inline .travis.yml:23]: These checks should be added in php 7.1

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,5 @@
+includes:
+	- vendor/jangregor/phpstan-prophecy/src/extension.neon
+parameters:
+    ignoreErrors:
+        - '#^Cannot cast object|string to string$#'
```
Comment (thomasvargiu) [inline phpstan.neon]: I think we shouldn't ignore this error. Where this error occurs?

File: src/Logger.php [hunk 3]
```diff
@@ -19,22 +19,16 @@ class Logger implements LoggerInterface
...
-    /**
-     * @var Raven_Client
-     */
+    /** @var Raven_Client */
     protected $client;
-    /**
-     * @var SenderInterface
-     */
+
+    /** @var Sender|SenderInterface */
```
Comment (thomasvargiu) [inline src/Logger.php]: We accept the interface in the constructor, so the implementation `Sender` should not be considered.

File: src/Logger.php [hunk 4]
```diff
@@ -19,22 +19,16 @@ class Logger implements LoggerInterface
...
     private $sender;
-    /**
-     * @var SanitizerInterface
-     */
+
+    /** @var Sanitizer|SanitizerInterface */
```
Comment (thomasvargiu) [inline src/Logger.php]: We accept the interface in the constructor, so the implementation `Sanitizer` should not be considered.

File: src/Logger.php (hunk 3 above)
Comment (thomasvargiu) [reply src/Logger.php]: Does phpstan throws errors?

File: src/Logger.php (hunk 4 above)
Comment (thomasvargiu) [reply src/Logger.php]: Does phpstan throws errors?

**Review by thomasvargiu — CHANGES_REQUESTED** [review]
Small changes... I think we should also add type hints where missing

File: phpstan.neon (hunk 2 above)
Comment (alexmanno) [reply phpstan.neon]: src/Logger.php:96

File: composer.json [hunk 5]
```diff
@@ -56,7 +58,8 @@
         ],
         "phpunit": "phpunit",
         "ci-phpunit": "phpunit --coverage-clover=coverage.clover",
-        "cs-check": "php-cs-fixer fix --dry-run --diff",
+        "ci-phpstan": "phpstan analyse --no-progress -l7 src tests",
```
Comment (thomasvargiu) [inline composer.json]: It's better to use current directory (`.`). Filters should be declared in phpstan configuration

File: src/Logger.php [hunk 6]
```diff
@@ -46,12 +40,17 @@ class Logger implements LoggerInterface
         LogLevel::DEBUG => Raven_Client::DEBUG,
     ];
 
-    const DEBUG = 'debug';
-    const INFO = 'info';
-    const WARN = 'warning';
-    const WARNING = 'warning';
-    const ERROR = 'error';
-    const FATAL = 'fatal';
+    public const DEBUG = 'debug';
```
Comment (thomasvargiu) [inline src/Logger.php]: These constants are useless. We can remove them.

**Review by thomasvargiu — CHANGES_REQUESTED** [review]

File: phpstan.neon (hunk 2 above)
Comment (thomasvargiu) [reply phpstan.neon]: Could be a good idea to do the following?

```php
if (\is_object($message) && \method_exists($object, '__toString')) {
    $message = $message->__toString();
}
```

File: .travis.yml [hunk 7]
```diff
@@ -15,17 +15,11 @@ env:
 matrix:
   fast_finish: true
   include:
-    - php: 7.0
-      env:
-        - DEPS=lowest
-    - php: 7.0
+    - php: 7.1
```
Comment (thomasvargiu) [inline .travis.yml:18]: There are still 2 tests with the same environment (php 7.1) with `DEPS=locked`.
`EXECUTE_CS_CHECK=true` and `EXECUTE_TEST_SCRUTINIZER=true` should be added in the already existing enviroment.

**Review by thomasvargiu — CHANGES_REQUESTED** [review]

**Review by thomasvargiu — APPROVED** [review]

**Pull request merged**